app/addons/documents/mango/mango.stores.js

```diff
@@ -131,21 +92,17 @@ Stores.MangoStore = FauxtonAPI.Store.extend({
         this.setQueryIndexCode(action.options);
       break;
 
-      case ActionTypes.MANGO_NEW_QUERY_FIND_CODE_FROM_FIELDS:
-        this.newQueryFindCodeFromFields(action.options);
-      break;
-
       case ActionTypes.MANGO_NEW_QUERY_FIND_CODE:
         this.setQueryFindCode(action.options);
       break;
 
-      case ActionTypes.MANGO_NEW_AVAILABLE_INDEXES:
-        this.setAvailableIndexes(action.options);
-      break;
+      case ActionTypes.MANGO_EXPLAIN_RESULTS:
+        this.setExplainPlan(action.options);
```

Comment:
Instead of passing the options object throught. Just pass the `explainPan`. Then you don't have to check that the options.explainPlan exists in the `setExplainPlan` function.